clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordBatch.java

```diff
@@ -67,10 +68,13 @@ public FutureRecordMetadata tryAppend(long timestamp, byte[] key, byte[] value,
         if (!this.records.hasRoomFor(key, value)) {
             return null;
         } else {
-            this.records.append(offsetCounter++, timestamp, key, value);
+            long checksum = Record.computeChecksum(timestamp, key, value, CompressionType.NONE, 0, -1);
+            int keyValueSize = (key == null ? 0 : key.length) + (value == null ? 0 : value.length);
+            this.records.append(offsetCounter++, timestamp, checksum, key, value);
```

Comment:
Instead of computing the CRC here and passing it into MemoryRecords, would it be better to still compute the CRC in Compressor.putRecord() and return it all the way to here? This way the computation of CRC is isolated in one place and there are fewer changes to constructors such as Record.